### PR TITLE
Add text export

### PR DESCRIPTION
**Pull request by harry-kim**
Allow user to export to text format

Refs: #SDE-3259

### CODE REVIEW

File: multi_import/multi_file_importer.py [hunk 1]
```diff
@@ -82,11 +84,23 @@ def pre_read(self, file_object):
         return file_object
 
 
+class TxtFormat(FileFormat):
+    def __init__(self):
+        super(TxtFormat, self).__init__(txt, read_file_as_string=True)
+
+    def write(self, dataset):
```
Comment (sibartlett) [inline multi_import/multi_file_importer.py]: I just checked the `_registerformats` method of `DataSet` [here](https://github.com/kennethreitz/tablib/blob/master/tablib/core.py#L252).

We should replace change the `write` method of `FileFormat` to the following:

```
def write(self, dataset):
    data = self.format.export_set(dataset)
    f = BytesIO()
    f.write(data)
    return f
```

Then we don't need `TxtFormat` class.

File: multi_import/_txt.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+    return stream.getvalue()
+
+
+def import_set(dset, in_stream, headers=True, **kwargs):
+    return NotImplementedError
+
+
+def detect(stream, delimiter=DEFAULT_DELIMITER):
+    return NotImplementedError
```
Comment (sibartlett) [inline multi_import/_txt.py]: This needs to return `False`, we must've raise an error here.

File: multi_import/multi_file_importer.py (hunk 1 above)
Comment (harry-kim) [inline multi_import/multi_file_importer.py]: updated

**Comment by lmontopo** [comment]
This is looking really good to me @harry-kim.  I'm assuming we will also update the app at some point to allow users to export in this format? 

Just a few minor things, really, but once those are addressed i'm ready to plus 1 this.

**Comment by harry-kim** [comment]
@lmontopo did you mean import? yes it will be added in the future.

**Comment by lmontopo** [comment]
No i actually did mean 'export' because even though you've now implemented the django_multi_import to handle exporting in this format, our app still doesn't have a radio button for 'export in txt'.  You know what i mean?  So as far as I can tell, the code you have written will never actually be used.  You know what i mean?

**Comment by harry-kim** [comment]
@lmontopo ah gotcha now. I've done that as well check MR #2225 on agra.  Once these are merged in together it should work.  If you want to test locally pip install -e this version and use my branch of SDE and you can export to text! :)

File: multi_import/_txt.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    stream = StringIO()
+
+    if not is_py3:
+        kwargs.setdefault('encoding', DEFAULT_ENCODING)
+
+    for row in dataset._package():
+        for key, val in row.iteritems():
+            stream.write('-' * len(key) + '\n')
+            stream.write(key.encode('utf-8') + '\n')
+            stream.write('-' * len(key) + '\n')
```
Comment (harry-kim) [inline multi_import/_txt.py]: yes, this was intentional so we get output like

`-------
task_id
-------
T1

------------
universal_id
------------
ff19485c-3607-48df-8269-5b0f0017c799
`

the * are for the section breaks.

Comment (lmontopo) [inline multi_import/_txt.py]: ah ok! thanks for explaining.

File: multi_import/_txt.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+from tablib.compat import is_py3, StringIO
+
+
+title = 'txt'
+extensions = ('txt',)
+
+
+DEFAULT_ENCODING = 'utf-8'
+DEFAULT_DELIMITER = ','
```
Comment (harry-kim) [inline multi_import/_txt.py]: removed

File: multi_import/_txt.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+
+def export_set(dataset, **kwargs):
+    """Returns CSV representation of Dataset."""
```
Comment (harry-kim) [inline multi_import/_txt.py]: changed

**Comment by lmontopo** [comment]
nice, looks good @harry-kim.  +1.

**Pull request merged**